src/main/java/org/wso2/carbon/securevault/hashicorp/config/HashiCorpVaultConfig.java

```diff
@@ -0,0 +1,142 @@
+import java.io.InputStream;
+import java.util.Properties;
+
+import static org.wso2.carbon.securevault.hashicorp.common.HashiCorpVaultConstants.ADDRESS_PARAMETER;
+import static org.wso2.carbon.securevault.hashicorp.common.HashiCorpVaultConstants.CONFIG_FILE_PATH;
+import static org.wso2.carbon.securevault.hashicorp.common.HashiCorpVaultConstants.DEFAULT_ENGINE_VERSION;
+import static org.wso2.carbon.securevault.hashicorp.common.HashiCorpVaultConstants.ENGINE_VERSION_PARAMETER;
+import static org.wso2.carbon.securevault.hashicorp.common.HashiCorpVaultConstants.NAMESPACE_PARAMETER;
+import static org.wso2.carbon.securevault.hashicorp.common.HashiCorpVaultConstants.ENGINE_PATH_PARAMETER;
+
+
+/**
+ * Configuration Loader for Vault Configurations.
+ */
+public class HashiCorpVaultConfig {
+
+    private static final Log log = LogFactory.getLog(HashiCorpVaultConfig.class);
+
+    private static String address;
+
+    private static String namespace;
+
+    private static String enginePath;
+
+    private static int engineVersion;
+
+    static {
+        load();
+    }
+
+    /**
+     * Load configurations.
+     */
+    public static void load() {
+
+        Properties properties = new Properties();
+        try (InputStream inputStream = new FileInputStream(CONFIG_FILE_PATH)) {
+            properties.load(inputStream);
+        } catch (IOException e) {
+            log.error("Error while loading configurations from " + CONFIG_FILE_PATH, e);
+        }
+
+        try {
+            address = getProperty(ADDRESS_PARAMETER, properties, true);
+            namespace = getProperty(NAMESPACE_PARAMETER, properties, false);
+            enginePath = getProperty(ENGINE_PATH_PARAMETER, properties, true);
+
+            String version = getProperty(ENGINE_VERSION_PARAMETER, properties, false);
+            engineVersion = version != null ? Integer.parseInt(version) : DEFAULT_ENGINE_VERSION;
+        } catch (HashiCorpVaultException e) {
+            log.error(e.getMessage(), e);
+        }
+
+    }
+
+    /**
+     * Get vault address.
+     *
+     * @return
+     */
+    public static String getAddress() {
+
+        return address;
+    }
+
+    /**
+     * Get namespace.
+     *
+     * @return
+     */
+    public static String getNamespace() {
+
+        return namespace;
+    }
+
+    /**
+     * Get engine path.
+     *
+     * @return
+     */
+    public static String getEnginePath() {
+
+        return enginePath;
+    }
+
+    /**
+     * Get vault engine version.
+     *
+     * @return
+     */
+    public static int getEngineVersion() {
+
+        return engineVersion;
+    }
+
+    /**
+     * Check existence and get property
+     *
+     * @param key Key
+     * @param properties Properties
+     * @param required Property is required
+     * @return
+     * @throws HashiCorpVaultException
+     */
+    private static String getProperty(String key, Properties properties, boolean required) throws HashiCorpVaultException {
```

Comment:
better if load only store the properties and have this method as the public interface,

at init phase of the caller class it will call,
config.load();

to read config it would call,
String address = config.getProperty(ADDRESS);

if it is required by the caller, they can null check and handle it rather than handling it here